### PR TITLE
File Converting

### LINKED ISSUE #25: File conversion
# Main idea
Commands for converting common file formats.
For example:
 - avif > jpg
 - mp4 > gif
 - png > bmp

The commands could be grouped depending on the file type (`/video convert`, `/image convert`, `/file convert` etc.), or be in one group (`/file convert`).

## Example usage:
`/video convert file:test.mp4 format:gif`
### Results in
`test.git` (as attachment)

### PR DESCRIPTION
**Pull request by filzaj**
For now supports: mp4 -> gif (other way around too) 
.png, .jpg, .webp, .bmp (You can convert from any of these)

### CODE REVIEW

**Comment by filzaj** [comment]
Everything should work, I'll add to it much later.

File: Program.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿using Discord;
+﻿﻿using Discord;
```
Comment (HEJOK254) [inline Program.cs]: What even changed here lol

File: Commands/Modules/VideoUtils.cs [hunk 2]
```diff
@@ -18,20 +18,19 @@ public async Task TrimVideoAsync(
 		[Summary(description: "A message to send with the video when it's trimmed")] string message = "",
 		[Summary(description: "If the video should be sent as a temporary message, that's only visible to you")] bool ephemeral = false)
 	{
-		string tempDirPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "tmp");
-		string videoInputPath = Path.Combine(tempDirPath, "input.mp4");      // Normally, I would use Path.GetTempFileName(), but FFMpegCore doesn't seem to
-		string videoOutputPath = Path.Combine(tempDirPath, "output.mp4");    // like the .tmp file extension (or anything other than .mp4) as far as i know
+		string videoInputPath = Path.GetTempFileName();
+		string videoOutputPath = Path.Combine(Path.GetTempPath(), videoInputPath + ".mp4");
```
Comment (HEJOK254) [inline Commands/Modules/VideoUtils.cs:22]: Shouldn't just `string videoOutputPath = videoInputPath + ".mp4"` be enough?

> Creates a uniquely named, zero-byte temporary file on disk and returns the full path of that file.
[From dotnet api docs](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=net-8.0)

File: Commands/Modules/Converter.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
+﻿using Discord;
+using Discord.Interactions;
+using FFMpegCore;
+using Serilog;
+using SixLabors.ImageSharp;
```
Comment (HEJOK254) [inline Commands/Modules/Converter.cs:5]: Gonna have to add the licence (apache 2.0)

Just putting it here so I don't forget (hopefully lol)

File: Commands/Modules/Converter.cs [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    [SlashCommand("convert", "Convert a file format to another one")]
+    public async Task ConvertAsync(
+        [Summary(description: "The video or image to convert")] Discord.Attachment attachment,
+        [Choice("gif", ".gif"), Choice("mp4", ".mp4"), Choice("mpg", ".mpg"), Choice("avi", ".avi"), Choice("mov", ".mov"), Choice("mkv", ".mkv"), Choice("png", ".png"), Choice("jpg", ".jpg"), Choice("webp", ".webp"), Choice("bmp", ".bmp"), Choice("tiff", ".tiff"), Choice("wmv", ".wmv"), Choice("flv", ".flv"), Choice("pbm", ".pbm"), Choice("tga", ".tga"), Choice("qoi", ".qoi")] string outputFormat,
+        [Summary("fps", description: "If using Gif, what fps should it have")] int fps = 30,
+        [Summary(description: "A message to send with the converted file")] string message = "",
+        [Summary(description: "If the file should be sent as a temporary message, that's only visible to you")] bool ephemeral = false)
+    {
+        string inputFilePath = Path.GetTempFileName();
+        string outputFilePath = Path.Combine(Path.GetTempPath(), inputFilePath + outputFormat);
```
Comment (HEJOK254) [inline Commands/Modules/Converter.cs]: Same thing as in `VideoUtils.cs 21-22`:

> Shouldn't just string videoOutputPath = videoInputPath + ".mp4" be enough?
> 
> >   Creates a uniquely named, zero-byte temporary file on disk and returns the full path of that file.
> >   [From dotnet api docs](https://learn.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=net-8.0)

File: Commands/Modules/Converter.cs [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                    await FollowupAsync($"Silly, you are converting from {Path.GetExtension(attachment.Filename)} to {outputFormat}.", ephemeral: ephemeral);
+                    return;
+                }
+                await converter(inputFilePath, outputFilePath, fps);
+            }
+            else
+            {
+                await FollowupAsync("Unsupported conversion type.", ephemeral: ephemeral);
+                return;
+            }
```
Comment (HEJOK254) [inline Commands/Modules/Converter.cs]: if could be inverted to reduce nesting
Not sure if this is more readable tho

```suggestion
            if (!conversionMap.TryGetValue(outputFormat, out var converter))
            {
                await FollowupAsync("Unsupported conversion type.", ephemeral: ephemeral);
                return;
            }

            if (Path.GetExtension(attachment.Filename) == outputFormat)
            {
                await FollowupAsync($"Silly, you are converting from {Path.GetExtension(attachment.Filename)} to {outputFormat}.", ephemeral: ephemeral);
                return;
            }
            await converter(inputFilePath, outputFilePath, fps);
```

**Review by HEJOK254 — COMMENTED** [review]
I'm gonna have to add the licence, but other than that, this looks good.
There's a few things that could be changed, though I'm not sure about them

**Review by HEJOK254 — APPROVED** [review]

**Pull request merged**